### PR TITLE
Catch session unavailable exception

### PR DESCRIPTION
**Pull request by DeepDiver1975**
port of https://github.com/owncloud/core/pull/30473

### CODE REVIEW

File: tests/lib/Session/CryptoSessionDataTest.php [hunk 1]
```diff
@@ -34,7 +36,7 @@ class CryptoSessionDataTest extends Session {
 	protected function setUp() {
 		parent::setUp();
 
-		$this->wrappedSession = new \OC\Session\Memory($this->getUniqueID());
+		$this->wrappedSession = new \OC\Session\Memory(static::getUniqueID());
```
Comment (PVince81) [inline tests/lib/Session/CryptoSessionDataTest.php]: never seen this `static::` approach before.

should this be `self::` ?

Comment (DeepDiver1975) [reply tests/lib/Session/CryptoSessionDataTest.php]: Maybe this helps to understand when to use what - https://www.leaseweb.com/labs/2014/04/static-versus-self-php/

In this case both can be used without changing the logic

Comment (PVince81) [reply tests/lib/Session/CryptoSessionDataTest.php]: I'm only asking when I see patterns we did not use much before. Fine to start using this now.

**Review by PVince81 — APPROVED** [review]
:+1:

**Pull request merged**